src/CornerstoneViewport/CornerstoneViewport.js

```diff
@@ -713,6 +724,96 @@ class CornerstoneViewport extends Component {
         );
       }
     }
+
+    // TODO? Should we shallow equality check these?
+    if (this.props.viewport !== prevProps.viewport) {
+      // Update the internal representation of the viewport parameters
+      let viewport = Object.assign(
+        {},
+        this.state.viewport,
+        this.props.viewport
+      );
+
+      // Handle reset and fitToWindow cases
+      // If viewport.scale === null or voi === null, call getDefaultViewportForImage
+      // and use these values prior to calling setViewport
+
+      const defaultViewport = cornerstone.getDefaultViewportForImage(
+        this.element,
+        cornerstone.getImage(this.element)
+      );
+      if (viewport.voi === null) {
+        viewport.voi = defaultViewport.voi;
+      }
+
+      if (viewport.zoomScale !== null) {
+        const maximumScale = 10;
+        const minimumScale = 0.05;
+
+        if (viewport.zoomScale === 0) {
+          viewport.scale = defaultViewport.scale;
+        } else if (viewport.zoomScale < 0) {
+          viewport.scale = Math.max(
+            viewport.scale + viewport.zoomScale,
+            minimumScale
+          );
+        } else {
+          viewport.scale = Math.min(
+            viewport.scale + viewport.zoomScale,
+            maximumScale
+          );
+        }
+      }
+
+      if (viewport.resetViewport) {
+        viewport = defaultViewport;
+      }
+
+      if (viewport.clearTools) {
+        const toolStateManager =
+          cornerstoneTools.globalImageIdSpecificToolStateManager;
+        toolStateManager.clear(this.element);
+        cornerstone.updateImage(this.element);
+      }
```

Comment:
is `clearTools` a bool? To trigger this and other bool props again, do you need to quickly flip them after this call?

Reply:
You mean every time the user clicks? If so, then yes. 

https://github.com/OHIF/ohif-core/blob/viewport/src/classes/HotkeysUtil.js#L135